pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -373,71 +374,5 @@ func (c *Cluster) addOrRemoveExternalMonitor(status client.MonStatusResponse) (b
 		logger.Debugf("everything is fine mon %s in the clusterInfo and its quorum status is %v", mon.Name, inQuorum)
 	}
 
-	// Let's do a new iteration, over ClusterInfo this time to catch mon that might have disappeared
```

Comment:
If this goes away how do we manage mon disappearing?

Reply:
If we want to remove all the steps from where we see if the MONs are in mon map, I was thinking that then the only check required to see if mons are present or disappeared is by checking `clusterinfo` and `quorum`.
That was the case right?.

Reply:
The goal here, is to always query the source of truth: the external cluster monitor map. We are getting it via `status.MonMap.Mons`. So what you need to do is to:

* reset clusterInfo each time a check is performed
*  loop against `range status.MonMap.Mons` and populate clusterInfo

Does that make sense?

Reply:
So the check gets performed at every 45 seconds.
Means I need to clear clusterinfo every 45 seconds and populate it again?.

Reply:
Yes.

Reply:
@travisn ?

Reply:
Agreed, every time we successfully connect to the remote cluster I would expect that we reset the clusterinfo based on that connection. If we fail to connect, we wouldn't reset it though. 
Separate question... is 45 seconds necessary? That seems very frequent.

Reply:
45s is the default health check, I don't mind keeping it.